src/cdk/platform/platform.ts

```diff
@@ -20,7 +21,8 @@ const hasV8BreakIterator = (typeof Intl !== 'undefined' && (Intl as any).v8Break
 @Injectable({providedIn: 'root'})
 export class Platform {
   /** Whether the Angular application is being rendered in the browser. */
-  isBrowser: boolean = typeof document === 'object' && !!document;
+  isBrowser: boolean = this._platformId ?
+    isPlatformBrowser(this._platformId) : typeof document === 'object' && !!document;
```

Comment:
I'm not sure I see the benefit of making this change. If someone wants to shim the `document` to pretend this is the browser, why not respect that?

Reply:
Because if they shim the document but not the navigator, like Domino does, all of the other checks result in errors

Reply:
Not to mention that it might not be an intentional shim

Reply:
You have to go through a bit more trouble to declare a global variable in Node though (e.g. `global.document = document`).

Reply:
Yeah, but it's happening to me. This PR didn't just come out of nowhere. The Angular testing utilities are polluting the global namespace for Node tests (see [here](https://github.com/angular/angular/blob/master/packages/core/testing/src/render3.ts#L88))

Reply:
Can you add some commenting here that explains that? Especially the part about what Domino does

Reply:
Sure, will add. Alternatively I can add a check for the navigator to each of the below checks. Let me know which you prefer

Reply:
Using the platform id works, I just want people who look at this in the future to be able to take advantage of all of the information you picked up while figuring this out